src/devices/MettlerToledo/MettlerToledoDevice.cs

```diff
@@ -0,0 +1,362 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Design;
+using System.IO.Ports;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+
+using Iot.Device.MettlerToledo.Exceptions;
+using Iot.Device.MettlerToledo.Messages;
+using Iot.Device.MettlerToledo.Readings;
+using UnitsNet;
+
+namespace Iot.Device.MettlerToledo
+{
+    /// <summary>
+    /// <see cref="MettlerToledoDevice.WeightUpdated"/>
+    /// </summary>
+    /// <param name="sender">Object where the event occurred.</param>
+    /// <param name="e">Event data.</param>
+    public delegate void WeightUpdatedHandler(object sender, MettlerToledoWeightReading e);
+
+    /// <summary>
+    /// Provides an interface for interacting with MT-SICS devices, like scales.
+    /// </summary>
+    public class MettlerToledoDevice : IDisposable
+    {
+        // TODO: we need to test on a real scale to determine what the response for other weights are when they are set to Unit 1
+        private static readonly (string mtAbbreviation, MettlerToledoWeightUnit mtUnit, UnitsNet.Units.MassUnit unitsNetUnit)[] _unitMapping =
+        {
+            ("g", MettlerToledoWeightUnit.Grams, UnitsNet.Units.MassUnit.Gram),
```

Comment:
Why are there only 3 enums mapped while you have more? Also seems that all should be able to be mapped with the MassUint.
The abbreviation can be gathered from the UnitsNet as well.

Reply:
So these are abbreviations given by the MT-SICS scales, which seem to be different than some of the normal abbreviations. Unfortunately without access to the physical scale I can't confirm what the abbreviations are yet (which is why only 3 are provided)